rbi/core/false_class.rbi

```diff
@@ -23,4 +23,12 @@ class FalseClass
   sig {returns(TrueClass)}
   def !
   end
+
+  sig {params(obj: FalseClass).returns(TrueClass)}
+  sig {params(obj: BasicObject).returns(FalseClass)}
+  def ==(obj); end
```

Comment:
For `T.untyped`, this selects `TrueClass`.

Reply:
(the blog post you link mentions overloaded signatures, and it seems that pay-server's codebase is failing for the same reasons outlined in the blog post)